### DISCUSSION TITLE
[IntelliSense] Compatibility with `editor.defaultColorDecorators` setting

### DISCUSSION THREAD

**Discussion by olets** [tailwindlabs/tailwindcss#11212] [General]
&nbsp;

> Update: now a VS Code feature request which you can thumbsup if you're so inspired https://github.com/microsoft/vscode/issues/182214

> Update 2: If you'd like to see the VS Code team work on this, give https://github.com/microsoft/vscode/issues/182214 a thumbs up. If it does not receive 20 thumbs up by July 22, it will be closed.

> Update 3: The above issue received enough thumbs ups to make it into their backlog.

&nbsp;

**What version of Tailwind CSS IntelliSense are you using?**

0.9.11

**What version of Tailwind CSS are you using?**

3.3.1

**What package manager are you using?**

npm

**What operating system are you using?**

macOS

**Tailwind config**

Lmk if this is really needed. Minimal config for reproduction:

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [/* fill me in */],
  theme: {
    extend: {
      colors: {
        example: '#ff0000',
      }
    }
  },
}
```

**VS Code settings**

Lmk if this is really needed.

**Reproduction URL**

Lmk if you have a recommended way of creating cloud reproductions for this plugin.

**Describe your issue**

VS Code 1.78 (April 2023) adds inline color decorators

> It is also now possible to visualize inline color decorators for CSS-formatted colors in all file types. To display these decorators, enable the Editor: Default Color Decorators (`editor.defaultColorDecorators`) setting.

(source: https://code.visualstudio.com/updates/v1_78#_standalone-color-picker)

When `editor.defaultColorDecorators` is set to `true:

- Without Tailwind CSS Intellisense enabled, color decorators appear. Screenshot example of the Tailwind config file (but the bug is not limited to this one file):
    ![tailwind config screenshot showing a red color decorator next to the color value ff0000](https://user-images.githubusercontent.com/3282350/236341098-6a75d1ac-f77d-4da9-97d0-4bec9863c47b.png)
- Without Tailwind CSS Intellisense enabled, color decorators appear in the Tailwind config file only until Tailwind CSS Intellisense loads, at which point they disappear.

**Comment by gyurielf** [comment]
I ran into this as well.
It breaks the html class colors previews too.
![image](https://user-images.githubusercontent.com/12744391/236510091-38690a21-723e-4445-aeb5-ea5802d3989e.png)

UPDATE
The color previews appears, but after 5-8 minutes, and the disappers randomly. (I made a single html boilerplate file for it)

**Reply by GBrachetta** [reply]
If it helps, I had installed the extension [json-color-token](https://marketplace.visualstudio.com/items?itemName=yechunan.json-color-token). Uninstalling it brought back the color previews in Tailwind classes.

**Comment by starball5** [comment]
Similar past issues with colour decorations not appearing: tailwindlabs/tailwindcss-intellisense#728, tailwindlabs/tailwindcss-intellisense#750. Though this issue seems instead to have to do with changes made in VS Code 1.78 (see https://code.visualstudio.com/updates/v1_78#_standalone-color-picker)

Potentially related on Stack Overflow: 

- [Why I can't see the color preview box in VS Code?](https://stackoverflow.com/q/76182741/11107541)
- [VScode doesnt show square with color using Tailwind](https://stackoverflow.com/q/76192636/11107541)

Both those users in the above linked questions are using version 0.9.11 of this extension with VS Code 1.78.0.

**Comment by bradlc** [comment]
Not really sure what to do here 🤔 As far as I can tell there's no way to have extension-provided colour decorators in addition to the new "default colour decorators" in the same document. That's just how this new VS Code feature works unfortunately. As soon as any extension provides colour decorators for a document the "default" decorators are disabled. I'm going to convert this to a discussion because I don't consider this a bug with the extension, but if anyone has any ideas for improving the colour decorator behaviour please let me know!

**Reply by olets** [reply]
- Thanks. I've opened a VS Code feature request https://github.com/microsoft/vscode/issues/182214

**Reply by olets** [reply]
The request has been accepted into the backlog 🎉

**Comment by starball5** [comment]
The default value of `editor.defaultColorDecorators` is `false`. @olets did you set it to `true`? Because if you didn't, then the value should be `false`, and @bradlc's previous comment might not be going in the right direction.

**Reply by olets** [reply]
This is when `true`. Updated the description to clarify

**Comment by seccentral** [comment]
Repeated testing with and without VSCode custom user settings (nuked the json file) are reproducible as detailed below 
Environment:  
<details>
<summary>Click to expand </summary>

vscode `1.78.2`
Tailwind CSS Intellisense `v0.9.11`
pnpm `8.5.1` 
node `v20.1.0`
Astro `2.4.1`
Nextjs `13.4.4`
</details> 

Color Boxes **`Work`** on NextJS: 
<details>
<summary> Click to expand </summary>

(Accept all prompts, just go for defaults everywhere)
```
pnpm create create-next-app@latest nextjscolors
cd nextjscolors
code . 
```
Results in this: 
![image](https://github.com/tailwindlabs/tailwindcss/assets/3766320/d12cd9fe-5469-46ce-aad3-982805fb9a31)
</details>

Color Boxes `Fail` on Astro: 
<details>
<summary> Click to expand </summary>

(Accept all prompts, just go for defaults everywhere)
```
pnpm create astro@latest astrocolors
cd astrocolors
pnpm astro add tailwind
code .
```
Results in this :
 ![image](https://github.com/tailwindlabs/tailwindcss/assets/3766320/d68aa992-d570-45e7-ac68-fb89c826750d)
</details> 

Color boxes `Fail` on tailwind vanilla: 
<details>
<summary> Click to expand </summary> 

I even did a standalone tailwind install, Still the same issue: 
![image](https://github.com/tailwindlabs/tailwindcss/assets/3766320/b33edef3-03af-4144-a971-96be9d7bd025)
</details>

There is clearly something happening in one environment that is missing in the other. Same results with npm. 

Whatever next is doing or (not) doing seems to work. Hope this helps someone get to the bottom of this, I'm still new at this and don't really know where to start debugging (except isolating environments like i've done above)
